### PR TITLE
Add Minecraft LCE save format

### PR DESCRIPTION
**Pull request by DexrnZacAttack**

### CODE REVIEW

**Comment by DexrnZacAttack** [comment]
Very simple for now but I'll probs update it tomorrow or something; it's 3 am right now.

**Comment by DexrnZacAttack** [comment]
updated to add files to pattern

**Comment by DexrnZacAttack** [comment]
NBT Parsing!

**Comment by WerWolv** [comment]
Hey, thanks a lot! 
Could you please also add your pattern to the table in the readme and add a test file as described in the PR template?

**Comment by DexrnZacAttack** [comment]
> Hey, thanks a lot! Could you please also add your pattern to the table in the readme and add a test file as described in the PR template?

Just saw this, sure!

File: patterns/lcesave.hexpat [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+#pragma disclaimer This pattern is not affiliated with Mojang Studios nor Microsoft. Minecraft is a trademark of Mojang Studios and Microsoft.
+#pragma endian big
+/*
+ * ^ switch this to little for Switch, PS4, or Xbox One
+ * so you'll probs have to switch all of the be to le
+*/
+
+// NOTE: This pattern was only tested on Nightly@974c4ba, things may break/not work on older versions.
+
+#include <std/string.pat>
```
Comment (C3pa) [inline patterns/lcesave.hexpat:13]: I believe that this line isn't needed in v1.35.4 of ImHex. The `import` line below should do just fine.

Comment (DexrnZacAttack) [reply patterns/lcesave.hexpat:13]: > I believe that this line isn't needed in v1.35.4 of ImHex. The `import` line below should do just fine.

Alright I'll probably fix that in my hexpat later, right now I'm trying to find a good file for the example folder and fix some issues I had with the NBT library (TypeTag not found iirc)

**Comment by DexrnZacAttack** [comment]
Finished updating, will put example files in soon!

**Comment by DexrnZacAttack** [comment]
Save files added! of course markdown doesn't work in descriptions for whatever reason so I will send here:

TU033 is a pre-release version, which has a slightly different file format that I documented myself [here](https://github.com/Team-Lodestone/Documentation/blob/main/LCE/Pre-Release%20Save%20Format.md)

TU5 is a release version which uses the same file format used all the way up to latest TU (aka the last version released)

All of these saves have been decompressed from XMemCompress (iirc modified LZX)
Chunk data however is still compressed, because it's not meant to be uncompressed in the scope of reading the save.

https://github.com/Team-Lodestone/Documentation/blob/main/LCE/LCE%20File%20Versions.md

**Comment by DexrnZacAttack** [comment]
**Should** be ready.

**Comment by applecuckoo** [comment]
pinging @WerWolv since he may have missed this

**Comment by WerWolv** [comment]
Thanks a lot!

**Comment by DexrnZacAttack** [comment]
> Thanks a lot!

Yeah!
How does updating the patterns work btw... Do I just make another pull request?

**Comment by WerWolv** [comment]
Exactly yeah

**Pull request merged**